### ISSUE TITLE
how to have a  remote debug use ghidra debug agent(GADP)

### ISSUE THREAD

**Issue by gjm-anban** [NationalSecurityAgency/ghidra#6016]
environment:
Ghidra_10.4_build
java17

my step is:
computer A: windows ,ghidra
computer B: linux on vm ,target binary
1、execute svrInstall on computer B,look at ./ghrdraSvr status 
![image](https://github.com/NationalSecurityAgency/ghidra/assets/73465248/f283513a-1bb6-4efb-ba7f-555427b5f8dd)
2、use ghidra debug agent on computer A to connect B,but always on connecting status .
![image](https://github.com/NationalSecurityAgency/ghidra/assets/73465248/fea4fcfd-9e0b-4e67-abc6-f282406f724a)

how can i do to debug on remote linux server.I have tried gdb via SSH ,but gdb connected for a split second before it broke. Tried gdb via gadp but return java.lang.RuntimeException: Agent terminated with error: BindException: Cannot assign requested address: bind

**Comment by d-millar** [comment]
@gjm-anban I think you have the wrong server here.  ghidraSvr is for multi-user projects and isn’t really related to the debugger.  Instead, try support/gdbGADPServerRun (details in https://github.com/NationalSecurityAgency/ghidra/tree/master/GhidraDocs/GhidraClass/Debugged
/B1-RemoteTargets.md).  

Not really sure why “gdb over SSH” failed.  Can you run gdb on your target VM via ssh (without Ghidra in the mix)?

**Comment by gjm-anban** [comment]
> @gjm-anban I think you have the wrong server here. ghidraSvr is for multi-user projects and isn’t really related to the debugger. Instead, try support/gdbGADPServerRun (details in https://github.com/NationalSecurityAgency/ghidra/tree/master/GhidraDocs/GhidraClass/Debugged /B1-RemoteTargets.md).
> 
> Not really sure why “gdb over SSH” failed. Can you run gdb on your target VM via ssh (without Ghidra in the mix)?

I ran support/gdbGADPServerRun --agent-args -g /usr/bin/gdb . but it can not find gdb
![image](https://github.com/NationalSecurityAgency/ghidra/assets/73465248/e95c26cc-13b0-4c9e-a67b-a6c44445134c)

**Comment by d-millar** [comment]
hmmm, definitely seems broken, but I think we have a temporary workaround for you - edit gdbGADPServerRun and change "--gadp-args" to "--agent-args".  That should at least get the server up and running.  The ip/port to use in "gdb using GADP" will be listed in the output from that shell command.

**Comment by nsadeveloper789** [comment]
@gjm-anban The quick fix is to change the last line of `support/gdbGADPServerRun`.
```bash
# Launch Ghidra
"${SCRIPT_DIR}"/launch.sh fg jdk GdbAgent "${MAXMEM}" "" agent.gdb.gadp.GdbGadpServerLaunchShim $@
```
It basically comes down to removing the hard-coded arguments and putting `$@` there instead. You should then be able to re-try your invocation: `support/gdbGADPServerRun --agent-args -g /usr/bin/gdb`

**Comment by ryanmkurtz** [comment]
Fixed by b6100233ddff2e1c30f9f4ca86d13ae26e8ddcec
